test/enclave_wrangler_tests/test_dataset_upload.py

```diff
@@ -0,0 +1,77 @@
+"""Tests
+
+How to run:
+    python -m unittest discover
+
+TODO's
+ - 1. Test framework: Current implementation is ad-hoc for purposes of development.
+ - 2. Change from validate to apply, or do both
+"""
+import os
+import pickle
+import sys
+import unittest
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, List, Union
+
+import pandas as pd
+from requests import Response
+from sqlalchemy.exc import IntegrityError
+
+from enclave_wrangler.dataset_upload import upload_new_cset_container_with_concepts_from_csv, \
+    upload_new_cset_version_with_concepts_from_csv
+
+TEST_DIR = os.path.dirname(__file__)
+PROJECT_ROOT = Path(TEST_DIR).parent
+# todo: why is this necessary in this case and almost never otherwise?
+# https://stackoverflow.com/questions/33862963/python-cant-find-my-module
+sys.path.insert(0, str(PROJECT_ROOT))
+
+TEST_INPUT_DIR = os.path.join(TEST_DIR, '../input', 'test_enclave_wrangler')
+TEST_SCHEMA = 'test_n3c'
+yesterday: str = (datetime.now() - timedelta(days=1)).isoformat() + 'Z'  # works: 2023-01-01T00:00:00.000Z
+
+class TestDatasetUpload(unittest.TestCase):
+    def test_upload_cset_container_from_csv(self):
+        """Test uploading a concept set container from CSV"""
+        # response: Response = upload_concept_set_container(
+        #     concept_set_id='x', intention='x', research_project='x', assigned_sme=user_id,
+        #     assigned_informatician=user_id)
+        # if not('result' in response and response['result'] == 'VALID'):
+        #     print('Failure: test_upload_concept_set\n', response, file=sys.stderr)
+        inpath = os.path.join(TEST_INPUT_DIR, 'test_upload_cset_container_from_csv', 'new_container.csv')
+        df = pd.read_csv(inpath).fillna('')
+        response: Dict = upload_new_cset_container_with_concepts_from_csv(df=df)
+        print(response)
+
+
+    # TODO #1: for "success case", these things that aren't in the DB yet will be, so will need to instead
+    #  (a) fetch recent objects, and just pick 1 if anything new exists, (b) (best) fetch junk concept set version etc
+    #  (c) We could pick an 'archived' concept set container. After we fetch it, turn the 'archived' flag off.
+
+    def _test_upload_cset_version_from_csv(self, path: str):
```

Comment:
For this one there was actually no green arrow, which I found quite confusing.

Reply:
Ah, that is because there is a `_` at the beginning of it. It only knows it's a test if it starts with `test_`. There was a reason I put a `_` in front of it... but I no longer remember why. Maybe it's better if it remains deactivated , since there are still some todo's in it.